bot/exts/halloween/spookynamerate.py

```diff
@@ -0,0 +1,409 @@
+
+            async with channel.typing():
+                await asyncio.sleep(1)  # give the drum roll feel
+
+                if not winners:  # There are no winners (no participants)
+                    await channel.send("Hmm... Looks like no one participated! :cry:")
+                    return
+
+                if len(winners) > 1:
+                    await channel.send(" and ".join(PING.format(id=win[1]["author"]) for win in winners) + " !")
+                    score = winners[0][1]["score"]
+                    # display winners, their names and scores
+                    await channel.send(f"Congratulations to all! You have a score of {score}!")
+                    names = ", ".join([f'{win[1]["name"]} ({PING.format(id=win[1]["author"])})' for win in winners])
+                    await channel.send(f"Your names were: {names}!")
+
+                else:  # there is only one winner
+                    # There cant be 0 winners since we are checking that above
+
+                    winner = winners[0][1]
+                    await channel.send(f"{PING.format(id=winner['author'])}!")
+                    await channel.send(
+                        f"Congratulations {PING.format(id=winner['author'])}! You have a score of "
+                        f"{winner['score']}!"
+                    )
+                    await channel.send(f"Your name was: **{winner['name']}**.")  # display winner, name and score
+
+                # Send random party emojis
+                party = (random.choice([":partying_face:", ":tada:"]) for _ in range(random.randint(1, 10)))
+                await channel.send(" ".join(party))
+
+            async with self.checking_messages:  # Acquire the lock to delete the messages
+                await self.messages.clear()  # reset the messages
+
+        # send the next name
+        self.name = f"{random.choice(self.first_names)} {random.choice(self.last_names)}"
+        await self.data.set("name", self.name)
+
+        await channel.send(
+            "Let's move on to the next name!\nAnd the next name is...\n"
+            f"**{self.name}**!\nTry to spookify that... :smirk:"
+        )
+
+        self.poll = False  # accepting responses
+
+    @announce_name.before_loop
+    async def wait_till_scheduled_time(self) -> None:
+        """Waits till the next day's 12AM if crossed it, otherwise waits till the same day's 12AM."""
+        if SpookyNameRate.debug:
+            return
+
+        now = datetime.utcnow()
+        if now.hour < 12:
+            twelve_pm = now.replace(hour=12, minute=0, second=0, microsecond=0)
+            time_left = twelve_pm - now
+            await asyncio.sleep(time_left.seconds)
+            return
+
+        tomorrow_12pm = now + timedelta(days=1)
+        tomorrow_12pm = tomorrow_12pm.replace(hour=12, minute=0, second=0, microsecond=0)
+        await asyncio.sleep((tomorrow_12pm - now).seconds)
```

Comment:
One says it's 12AM, code says 12PM. I'd much more prefer if it either said midnight or midday, and be the same

Reply:
Oops I'll change the docstring, I wanted it to be midday, I got confused